toolkit/tools/liveinstaller/liveinstaller.go

```diff
@@ -119,19 +125,69 @@ func installerFactory(forceAttended bool, configFile, templateConfigFile string)
 	}
 
 	if isAttended {
-		installFunc = func(args imagerArguments) (bool, error) {
+		installFunc = func(args imagerArguments) (installationDetails, error) {
 			return terminalUIAttendedInstall(templateConfigFile, args)
 		}
 	} else {
-		installFunc = func(args imagerArguments) (bool, error) {
+		installFunc = func(args imagerArguments) (installationDetails, error) {
 			return unattendedInstall(configFile, args)
 		}
 	}
 
 	return
 }
 
-func updateBootOrder() (err error) {
+func updateBootOrder(installDetails installationDetails) (err error) {
+	if (installDetails.finalConfig.DefaultSystemConfig.BootType != "efi") {
+		logger.Log.Info("No BootType of 'efi' detected. Not attempting to set EFI boot order.")
+		return
+	}
+
+	err = removeOldMarinerBootTargets()
```

Comment:
good point! I will refactor so that they both make calls to shell.executelive() inside of their function body. Removing this awkward code splitting that's done with the Creation command! Thank you Cameron!